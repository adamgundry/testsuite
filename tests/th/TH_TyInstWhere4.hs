{-# LANGUAGE PolyKinds, DataKinds, TemplateHaskell, TypeFamilies #-}

module TH_TyInstWhere4 where

import Language.Haskell.TH

type family F a b :: Bool
type instance where
  F a a = True
  F a b = False

$( do { info1 <- reify ''F
      ; reportWarning (pprint info1)
      ; info2 <- reifyInstances ''F [ConT ''Int, ConT ''Int]
      ; reportWarning (pprint info2)
      ; info3 <- reifyInstances ''F [ConT ''Int, ConT ''Bool]
      ; reportWarning (pprint info3)
      ; return [] })


