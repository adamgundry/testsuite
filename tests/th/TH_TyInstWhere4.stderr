
TH_TyInstWhere4.hs:12:4: Warning:
    type family TH_TyInstWhere4.F a_0 b_1 :: * -> * -> GHC.Types.Bool
type instance where
    TH_TyInstWhere4.F a_2 a_2 = GHC.Types.True
    TH_TyInstWhere4.F a_3 b_4 = GHC.Types.False

TH_TyInstWhere4.hs:12:4: Warning:
    type instance where
    TH_TyInstWhere4.F a_0 a_0 = GHC.Types.True
    TH_TyInstWhere4.F a_1 b_2 = GHC.Types.False

TH_TyInstWhere4.hs:12:4: Warning:
    type instance where
    TH_TyInstWhere4.F a_0 a_0 = GHC.Types.True
    TH_TyInstWhere4.F a_1 b_2 = GHC.Types.False
